src/main/scala/com/fulcrumgenomics/util/SampleBarcodeMetric.scala

```diff
@@ -145,16 +156,26 @@ case class SampleBarcodeMetric
   var pf_perfect_matches: Metric.Count                         = 0,
   var one_mismatch_matches: Metric.Count                       = 0,
   var pf_one_mismatch_matches: Metric.Count                    = 0,
+  var bases_with_q_above_threshold: Metric.Count               = 0,
+  var total_number_of_bases: Metric.Count                      = 0,
   var fraction_matches: Metric.Proportion                      = 0d,
   var ratio_this_barcode_to_best_barcode: Metric.Proportion    = 0d,
   var pf_fraction_matches: Metric.Proportion                   = 0d,
   var pf_ratio_this_barcode_to_best_barcode: Metric.Proportion = 0d,
-  var pf_normalized_matches: Metric.Proportion                 = 0d
+  var pf_normalized_matches: Metric.Proportion                 = 0d,
+  var fraction_bases_above_threshold: Metric.Proportion        = 0d,
 ) extends Metric {
 
-  /** Increments the counts for the metric. */
-  def increment(numMismatches: Int, isPf: Boolean = true): Unit = {
 
+  /** Increments the counts for a metric
+    *
+    * @param numMismatches number of mismatches
+    * @param isPf true if the template passes QC
+    * @param basesToAdd number of total bases in the record
+    * @param qualitiesPass number of bases that have a quality score of 30 or higher
+    * @param omitFailing if failing reads are to be omitted from the output
+    */
+  def increment(numMismatches: Int, isPf: Boolean = true, basesToAdd: Int, qualitiesPass: Int, omitFailing: Boolean): Unit = {
```

Comment:
alternatively to having the logic on `isPf || !omitFailing` below, you could do one of the following:
1. `omitFailing` could be `countTowardsTotals`
2. or, `basesToAdd` and `qualitiesPass` would be provided as `0` by the caller, so `omitFailing` is not needed